src/ashish_Vyas/ShopSystem.java

```diff
@@ -0,0 +1,122 @@
+package ashish_Vyas;
+
+public class ShopSystem {
+	int maggiePackets;
+	int dosaPackets;
+	int oilPouches; 
+	int panipuriPackets;
+	int masalaPackets;
+	
+	void setInitialvalues(int maggie, int dosa, int oil, int panipuri, int masala) {
+		maggiePackets = maggie;
+		dosaPackets = dosa;
+		oilPouches = oil;
+		panipuriPackets = panipuri;
+		masalaPackets = masala;
+	}
+	
+	void requestedQ(int reqMaggie, int reqDosa, int reqOil, int reqPanipuri, int reqMasala, String pAllowedStatus) {
+		//Quantity Calculation for Maggie
+		if(reqMaggie <= maggiePackets)
+			maggiePackets= maggiePackets - reqMaggie;
+		else if (maggiePackets == 0) {
```

Comment:
we will not show OOS here, we have separate requirement for this right FR2 and FR3